src/sdk/sts_sdk.js

```diff
@@ -73,7 +73,7 @@ class StsSDK {
                 }
             }
             const signature = signature_utils.get_signature_from_auth_token(
-                token, this.requesting_account.access_keys[0].secret_key.unwrap());
+                token, token.temp_secret_key || this.requesting_account.access_keys[0].secret_key.unwrap());
```

Comment:
In the ```auth_server``` you've used an intermediate variable for that, is there a reason not to use it here likewise?

Reply:
added a variable